src/NServiceBus.Hosting.Windows/StartableAndStoppable/StartableAndStoppableBag.cs

```diff
@@ -0,0 +1,55 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using ObjectBuilder;
+
+    class StartableAndStoppableBag : IHoldStartAndStoppables
+    {
+        ConcurrentBag<IWantToRunWhenEndpointStartsAndStops> explicitInstances = new ConcurrentBag<IWantToRunWhenEndpointStartsAndStops>();
+        List<Type> scannedStartAndStoppableTypes;
+
+        public StartableAndStoppableBag(IEnumerable<Assembly> assemblies)
+        {
+            scannedStartAndStoppableTypes =
+                assemblies
+                    .SelectMany(assembly => assembly.GetTypes())
+                    .Where(IsConcrete)
+                    .Where(IsIWantToRunWhenEndpointStartsAndStops)
+                    .ToList();
+        }
+
+        public void Add(IWantToRunWhenEndpointStartsAndStops instance)
+        {
+            explicitInstances.Add(instance);
+        }
+
+        public IEnumerable<IWantToRunWhenEndpointStartsAndStops> Resolve(IConfigureComponents configurer, IBuilder builder)
+        {
+            foreach (var startAndStoppableType in scannedStartAndStoppableTypes)
+            {
+                configurer.ConfigureComponent(startAndStoppableType, DependencyLifecycle.InstancePerCall);
```

Comment:
Yeah, I could rename have another method specifically for register. No problem there.